client/jetpack-connect/jetpack-connection.jsx

```diff
@@ -90,11 +98,10 @@ const jetpackConnection = ( WrappedComponent ) => {
 				if ( currentPlan ) {
 					if ( currentPlan === PLAN_JETPACK_FREE ) {
 						debug( `Redirecting to wpadmin` );
-						externalRedirect( this.props.siteHomeUrl + JETPACK_ADMIN_PATH );
-					} else {
-						debug( `Redirecting to checkout with ${ currentPlan } plan retrieved from cookies` );
-						this.redirect( 'checkout', url, currentPlan, queryArgs );
+						return externalRedirect( this.props.siteHomeUrl + JETPACK_ADMIN_PATH );
```

Comment:
We don't need to wait for the rest of the controller to be executed if we are redirecting the user to an external route. Better to return early.